cedar-lean/Cedar/Thm/Data/List.lean

```diff
@@ -1285,23 +1264,21 @@ theorem foldlM_of_assoc_some (f : α → α → Option α) (x₀ x₁ x₂ x₃
   (h₃ : List.foldlM f x₂ xs = some x₃):
   (do let y ← List.foldlM f x₁ xs ; f x₀ y) = some x₃
 := by
-  cases xs
+  cases xs <;> csimp at *
   case nil =>
-    simp only [Option.bind_eq_bind, List.foldlM, pure, Option.some.injEq, Option.bind_some_fun] at *
     subst h₃; exact h₂
   case cons hd tl =>
-    simp only [Option.bind_eq_bind, List.foldlM, Option.bind_eq_some] at *
-    cases h₄ : f x₂ hd <;> simp only [h₄, false_and, exists_false, Option.some.injEq, exists_eq_left'] at h₃
+    cases h₄ : f x₂ hd <;> rw [h₄] at h₃ <;> csimp at h₃
```

Comment:
Here's an example where I think `csimp` unquestionably improves readability.  It also calls out much more clearly the one non-`simp` lemma (h4) which was previously used in these three long `simp only` invocations.

Reply:
I honestly think that the improvement in readability is minimal here.